src/neuroglancer/ui/segment_select_tools.ts

```diff
@@ -0,0 +1,131 @@
+});
+
+enum ToolState {
+  IGNORE,
+  SELECT,
+  DESELECT,
+};
+
+export class SelectSegmentsTool extends Tool<SegmentationUserLayer> {
+  constructor(layer: SegmentationUserLayer) {
+    super(layer);
+  }
+
+  toJSON() {
+    return SELECT_SEGMENTS_TOOLS_ID;
+  }
+
+  activate(activation: ToolActivation<this>) {
+    const {layer} = this;
+    const {body, header} = makeToolActivationStatusMessageWithHeader(activation);
+    let state = ToolState.IGNORE;
+    activation.bindInputEventMap(SELECT_SEGMENTS_INPUT_EVENT_MAP);
+    const updateStatus = () => {
+      removeChildren(body);
+      const msg = document.createElement('span');
+      switch (state) {
+        case ToolState.IGNORE:
+          header.textContent = 'Select/Deselect segments';
+          msg.textContent = `${selectEvent} to select segments; ${deselectEvent} to deselect segments.`;
+          break;
+        case ToolState.SELECT:
+        case ToolState.DESELECT:
+          header.textContent = `${state == ToolState.SELECT ? 'Select' : 'Deselect'} segments`;
+          msg.textContent = `Drag to ${state == ToolState.SELECT ? 'select' : 'deselect'} segments (${layer.displayState.segmentationGroupState.value.visibleSegments.size} selected).`;
+      }
+      body.appendChild(msg);
+    };
+    updateStatus();
+
+    const trySelectSegment = () => {
+      if (state == ToolState.IGNORE) {
+        return;
+      }
+      const {segmentSelectionState} = layer.displayState;
+      if (segmentSelectionState.hasSelectedSegment) {
+        const segment = segmentSelectionState.selectedSegment;
+        const {visibleSegments} = layer.displayState.segmentationGroupState.value;
+        switch (state) {
+          case ToolState.SELECT:
+            visibleSegments.add(segment);
+            break;
+          case ToolState.DESELECT:
+            visibleSegments.delete(segment);
+            break;
+        }
+      }
+    };
+
+    const startSelecting = (event: ActionEvent<MouseEvent>) => {
+      updateStatus();
+      trySelectSegment();
+      activation.registerDisposer(
+        layer.displayState.segmentSelectionState.changed.add(trySelectSegment));
+      activation.registerDisposer(
+        layer.displayState.segmentationGroupState.value.visibleSegments.changed.add(updateStatus));
+      startRelativeMouseDrag(
+        event.detail,
+        (_event, _deltaX, _deltaY) => {},
+        (_event) => {
+          state = ToolState.IGNORE;
+          updateStatus();
+        }
+      );
+    };
+
+    activation.bindAction('drag-select-segments', (event: ActionEvent<MouseEvent>) => {
+      event.stopPropagation();
+      state = ToolState.SELECT;
+      startSelecting(event);
+    });
+    activation.bindAction('drag-deselect-segments', (event: ActionEvent<MouseEvent>) => {
+      event.stopPropagation();
+      state = ToolState.DESELECT;
+      startSelecting(event);
+    });
+  }
+
+  get description() {
+    return 'select/deselect segments';
```

Comment:
Since this is shown in the top bar maybe make it more concise, e.g. just `select`.